src/Entity/Transaction.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity(repositoryClass="AppBundle\Repository\TransactionRepository")
+ */
+class Transaction
+{
+    public const PAYBOX_SUCCESS = '00000';
+    public const PAYBOX_CONNECTION_FAILED = '00001';
+    public const PAYBOX_INTERNAL_ERROR = '00003';
+    public const PAYBOX_CARD_NUMBER_INVALID = '00004';
+    public const PAYBOX_CARD_END_DATE_INVALID = '00008';
+    public const PAYBOX_CARD_UNAUTHORIZED = '000021';
+    public const PAYBOX_PAYMENT_PAGE_TIMEOUT = '000030';
+
+    /**
+     * @var int
+     *
+     * @ORM\Column(type="integer", options={"unsigned"=true})
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=100, nullable=true)
+     */
+    private $payboxResultCode;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=100, nullable=true)
+     */
+    private $payboxAuthorizationCode;
+
+    /**
+     * @var array|null
+     *
+     * @ORM\Column(type="json_array", nullable=true)
+     */
+    private $payboxPayload;
+
+    /**
+     * @var \DateTimeImmutable|null
+     *
+     * @ORM\Column(type="datetime_immutable", nullable=true)
+     */
+    private $payboxDateTime;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(unique=true, nullable=true)
+     */
+    private $payboxTransactionId;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(nullable=true)
+     */
+    private $payboxSubscriptionId;
+
+    /**
+     * @var \DateTimeImmutable
+     *
+     * @ORM\Column(type="datetime_immutable")
+     */
+    private $createdAt;
+
+    /**
+     * @var Donation
+     *
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Donation", inversedBy="transactions")
+     */
+    private $donation;
```

Comment:
We want to avoid  bidirectional association whenever it's possible so please make sure it's needed in both places.

https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/reference/best-practices.html#constrain-relationships-as-much-as-possible